src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -86,7 +86,7 @@ size_t MonitorList::unlink_deflated(Thread* self, LogStream* ls,
   ObjectMonitor* prev = NULL;
   ObjectMonitor* head = Atomic::load_acquire(&_head);
   ObjectMonitor* m = head;
-  while (m != NULL) {
+  do {
```

Comment:
This wasn't the loop I was referring to. It is the while loop below this at line 93.

Reply:
The inner while loop is now converted into a do-while loop.